Apps/Sandcastle/CesiumSandcastle.js

```diff
@@ -479,6 +481,16 @@ require({
         }
     });
 
+    window.onbeforeunload = function (e) {
+        var htmlText = (htmlEditor.getValue()).replace(/\s/g, '');
+        var jsText = (jsEditor.getValue()).replace(/\s/g, '');
+        if (demoHtml === htmlText && demoJs === jsText) {
+            return ;
+        }
+
+        return 'Changes made by you are not saved.';
```

Comment:
This function would be cleaner if you negate the if statement and move the return into it.  I would also update the message to something like "You have unsaved changes. Are you sure you want to navigate away from this demo?".

```javascript
if (demoHtml !== htmlText || demoJs !== jsText) {
    return 'You have unsaved changes. Are you sure you want to navigate away from this demo?';
}
```